modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/UpdatableTableImpl.java

```diff
@@ -281,7 +281,7 @@ private <RowT> BinaryRowEx convertRow(RowT row, ExecutionContext<RowT> ectx, boo
                 continue;
             }
 
-            Object value = hnd.get(colDesc.logicalIndex(), row);
+            Object value = hnd.get(keyOnly ? colDesc.physicalIndex() : colDesc.logicalIndex(), row);
```

Comment:
can we union these 2 loops : for (ColumnDescriptor colDesc : columnsOrderedByPhysSchema) { ?

Reply:
Thanks, removed first loop.